code/_onclick/item_attack.dm

```diff
@@ -236,11 +236,20 @@
 						power = K.defense(power,def_zone)
 					M.take_organ_damage(0, power)
 					to_chat(M, "Aargh it burns!")
+
+	//Break the item if applicable.
+	if(power && (I.breakable_flags & BREAKABLE_AS_MELEE) && !(I.breakable_flags & BREAKABLE_NOMOB))
+		take_damage(min(power, BREAKARMOR_MEDIUM), FALSE) //Cap recoil damage at BREAKARMOR_MEDIUM to avoid a powerful weapon also needing really strong armor to avoid breaking apart when used. Be verbose about the item being damaged if applicable.
+		break_item(null, null, null, null, null, M) //Break the item and spill any reagents onto the target.
```

Comment:
This is quite hard to read. You can use named arguments like so:
```suggestion
		break_item(hit_atom = M) //Break the item and spill any reagents onto the target.
```
The rest of the arguments will be null.

Reply:
That's good to know.